librz/bin/bin.c

```diff
@@ -1316,188 +1317,74 @@ RZ_API const RzBinXtrPlugin *rz_bin_xtrplugin_get(RZ_NONNULL RzBin *bin, RZ_NONN
 	return NULL;
 }
 
-static char *bin_demangle_cxx(RzBinFile *bf, const char *symbol, ut64 vaddr) {
-	char *out = rz_demangler_cxx(symbol);
-	if (!out || !bf) {
-		return out;
-	}
-	char *sign = (char *)strchr(out, '(');
-	if (!sign) {
-		return out;
-	}
-
-	char *str = out;
-	char *ptr = NULL;
-	char *method_name = NULL;
-	for (;;) {
-		ptr = strstr(str, "::");
-		if (!ptr || ptr > sign) {
-			break;
-		}
-		method_name = ptr;
-		str = ptr + 1;
-	}
-
-	if (RZ_STR_ISEMPTY(method_name)) {
-		return out;
-	}
-
-	*method_name = 0;
-	RzBinSymbol *sym = rz_bin_file_add_method(bf, out, method_name + 2, 0);
-	if (sym) {
-		if (sym->vaddr != 0 && sym->vaddr != vaddr) {
-			RZ_LOG_INFO("Duplicated method found: %s\n", sym->name);
-		}
-		if (sym->vaddr == 0) {
-			sym->vaddr = vaddr;
-		}
-	}
-	*method_name = ':';
-	return out;
-}
-
-static char *bin_demangle_rust(RzBinFile *bf, const char *symbol, ut64 vaddr) {
-	char *demangled = rz_demangler_rust(symbol);
-	if (!demangled) {
-		return demangled;
-	}
-
-	char *str = demangled;
-	char *ptr = NULL;
-	char *name = NULL;
-	while ((ptr = strstr(str, "::"))) {
-		name = ptr;
-		str = ptr + 2;
-	}
-
-	if (!name || RZ_STR_ISEMPTY(name + 2)) {
-		return demangled;
-	}
-
-	*name = 0;
-	RzBinSymbol *sym = rz_bin_file_add_method(bf, demangled, name + 2, 0);
-	if (sym && sym->vaddr == 0) {
-		sym->vaddr = vaddr;
-	}
-	*name = ':';
-	return demangled;
-}
-
 /**
- * \brief Demangles a symbol based on the language or the RzBinFile data
+ * \brief Demangles a symbol based on the language or by iterating all demanglers.
```

Comment:
For all `RzBinFiles` that are in the `RzBin` or just a current pne? Please specify that in a doxygen doc

Reply:
`RzBinFiles` is not used anymore. there was no need, it was a huge hack to add classes and methods.
This function should do one and only thing: demangle symbols.